model/flow/ledger.go

```diff
@@ -181,6 +181,18 @@ func (d RegisterEntries) Values() []RegisterValue {
 	return r
 }
 
+// RegisterIterator is used for iterating over a set of registers
+type RegisterIterator interface {
+	// First returns the first item of the iterator
+	First() *RegisterEntry
```

Comment:
rm.  HasNext should always be check first since the iterator yield an empty list